nym-api/src/network/handlers.rs

```diff
@@ -6,16 +6,29 @@ use crate::support::http::state::AppState;
 use axum::{extract, Router};
 use nym_contracts_common::ContractBuildInformation;
 use std::collections::HashMap;
+use tower_http::compression::CompressionLayer;
 use utoipa::ToSchema;
 
 pub(crate) fn nym_network_routes() -> Router<AppState> {
+    // This should compress responses unless:
+    //
+    // * They’re gRPC, which has its own protocol specific compression scheme.
+    // * It’s an image as determined by the content-type starting with image/.
+    // * They’re Server-Sent Events (SSE) as determined by the content-type being text/event-stream.
+    // * The response is less than 32 bytes.
+    let compression_layer: CompressionLayer = CompressionLayer::new()
```

Comment:
given you're creating this compression layer in 4 different spots, would it perhaps make sense to create a helper method for it instead? So that if we decide to change one of the settings, we wouldn't accidentally miss one of the instances.
Perhaps you could put it into `http-api-common/middleware` so that other APIs could benefit from that too